modules/cnf-performing-end-to-end-tests-running-hwlatdetect.adoc

```diff
@@ -0,0 +1,228 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/cnf-performing-platform-verification-latency-tests.adoc
+
+:_content-type: CONCEPT
+[id="cnf-performing-end-to-end-tests-running-hwlatdetect_{context}"]
+= Running hwlatdetect
+
+The `hwlatdetect` tool is available in the `rt-kernel` package with a regular subscription of {op-system-base-full} {op-system-version}.
+
+[IMPORTANT]
+====
+**Always** run the latency tests with `DISCOVERY_MODE=true` set. If you don't, the test suite will make changes to the running cluster configuration.
+====
+
+[NOTE]
+====
+When executing `podman` commands as a non-root or non-privileged user, mounting paths can fail with `permission denied` errors. To make the `podman` command work, append `:Z` to the volumes creation; for example, `-v $(pwd)/:/kubeconfig:Z`. This allows `podman` to do the proper SELinux relabeling.
+====
+
+.Prerequisites
+
+* You have installed the real-time kernel in the cluster.
+
+* You have logged in to `registry.redhat.io` with your Customer Portal credentials.
+
+.Procedure
+
+* To run the `hwlatdetect` tests, run the following command, substituting variable values as appropriate:
++
+[source,terminal,subs="attributes+"]
+----
+$ podman run -v $(pwd)/:/kubeconfig:Z -e KUBECONFIG=/kubeconfig/kubeconfig \
+-e LATENCY_TEST_RUN=true -e DISCOVERY_MODE=true -e ROLE_WORKER_CNF=worker-cnf \
+-e LATENCY_TEST_RUNTIME=600 -e MAXIMUM_LATENCY=20 \
+registry.redhat.io/openshift4/cnf-tests-rhel8:v{product-version} \
+/usr/bin/test-run.sh -ginkgo.v -ginkgo.focus="hwlatdetect"
+----
++
+The `hwlatdetect` test runs for 10 minutes (600 seconds). The test runs successfully when the maximum observed latency is lower than `MAXIMUM_LATENCY` (20 μs).
++
+If the results exceed the latency threshold, the test fails.
++
+[IMPORTANT]
+====
+For valid results, the test should run for at least 12 hours.
+====
++
+.Example failure output
+[source,terminal]
+----
+running /usr/bin/validationsuite -ginkgo.v -ginkgo.focus=hwlatdetect
+I0210 17:08:38.607699       7 request.go:668] Waited for 1.047200253s due to client-side throttling, not priority and fairness, request: GET:https://api.ocp.demo.lab:6443/apis/apps.openshift.io/v1?timeout=32s
+Running Suite: CNF Features e2e validation
+==========================================
+Random Seed: 1644512917
```

Comment:
Is the output of the commands copied from 4.10 docs? 
if so, I believe it isn't the right way to do that, but instead to retain the output we already have in the docs for each version, because different versions may have different outputs. Not a big deal, but just to not confuse the customer with unmatching output.

Reply:
The output is copied from 4.10, but I am almost certain that the output was not updated from when the docs were originally produced in 4.6. I will check.

Reply:
4.8-4.9 output == 4.10 output. So unless we want to run against 4.8 and 4.9 clusters I think we are OK. WDYT @shajmakh. I'll make your other suggestions thanks.